### PR TITLE
[Tests-Only] API register to list file in directory

### PR DESCRIPTION
**Pull request by swoichha**
## Description
API fucntion `listFile` has been added in ServerFiles.php to list out files in a given directory. Also it has been registered in routes.php. 

## Checklist:

- [ ] Latest changes are published according to [instructions in README.md](https://github.com/owncloud/testing#publish-latest-version-as-github-release)

### CODE REVIEW

**Comment by phil-davis** [comment]
@swoichha please pull latest testing app master, rebase this branch and force push.
It will be good to have the CI run against the very latest testing app code.

File: lib/ServerFiles.php [hunk 1]
```diff
@@ -199,4 +199,36 @@ public function deleteFile() {
 
 		return new Result();
 	}
+
+	/**
+	 * List file in the specified directory under the server root
+	 *'file' is the file to delete, including path from the server root
```
Comment (phil-davis) [inline lib/ServerFiles.php]: delete this line?

File: lib/ServerFiles.php [hunk 2]
```diff
@@ -199,4 +199,36 @@ public function deleteFile() {
 
 		return new Result();
 	}
+
+	/**
+	 * List file in the specified directory under the server root
```
Comment (phil-davis) [inline lib/ServerFiles.php]:
```suggestion
	 * List files in the specified directory under the server root
```
IMO it returns a list of all the files in the directory.

File: lib/ServerFiles.php [hunk 3]
```diff
@@ -199,4 +199,36 @@ public function deleteFile() {
...
+	 *'file' is the file to delete, including path from the server root
+	 * 'dir' is the directory to check, which may be inside other directories
+	 * e.g. 'apps2/myapp/appinfo'
+	 *
+	 * @return Result
+	 */
+	public function listFile() {
```
Comment (phil-davis) [inline lib/ServerFiles.php]:
```suggestion
	public function listFiles() {
```
Call this method `listFiles` ?

File: lib/ServerFiles.php [hunk 4]
```diff
@@ -199,4 +199,36 @@ public function deleteFile() {
...
+		$dir = \trim($this->request->getParam('dir'), '/');
+		if ($dir === "") {
+			return new Result(null, 400, "cannot list file in dir, no dir name given");
```
Comment (phil-davis) [inline lib/ServerFiles.php]:
```suggestion
			return new Result(null, 400, "cannot list files in dir, no dir name given");
```

**Comment by kiranparajuli589** [comment]
Lint failing...@swoichha try fixing linting errors!

File: lib/ServerFiles.php (hunk 3 above)
Comment (swoichha) [reply lib/ServerFiles.php]: Okay

**Comment by phil-davis** [comment]
Hint: `make test-php-style-fix`

**Comment by phil-davis** [comment]
@swoichha I have done the steps to "publish" this. You should be able to run CI for whatever PR needs this.

**Pull request merged**